models/Asset/Image/Thumbnail.php

```diff
@@ -112,18 +112,20 @@ protected function useOriginalFile(string $filename): bool
      */
     public function generate(bool $deferredAllowed = true): void
     {
-        if (!$this->checkAllowedFormats($this->config->getFormat(), $this->asset)) {
-            throw new ThumbnailFormatNotSupportedException();
-        }
-
-        if (!$this->checkMaxScalingFactor($this->config->getHighResolution())) {
-            throw new ThumbnailMaxScalingFactorException();
-        }
-
         $deferred = false;
         $generated = false;
 
         if ($this->asset && empty($this->pathReference)) {
+
+
+            if ($this->config && !$this->checkAllowedFormats($this->config->getFormat(), $this->asset)) {
```

Comment:
Can be moved to the else block in line 136 without config check, right?

Reply:
I already tried that, dealing with "cognitive overload" and qodana currently, trying to resolve it :) but thanks